test/bigtable/table.js

```diff
@@ -0,0 +1,852 @@
+  before(function() {
+    mockery.registerMock(
+      '../../lib/common/grpc-service-object', FakeGrpcServiceObject);
+    mockery.registerMock('../../lib/bigtable/family', FakeFamily);
+    mockery.registerMock('../../lib/bigtable/mutation', FakeMutation);
+    mockery.registerMock('../../lib/bigtable/filter', FakeFilter);
+    mockery.registerMock('pumpify', pumpify);
+    mockery.registerMock('../../lib/bigtable/row', FakeRow);
+
+    mockery.enable({
+      useCleanCache: true,
+      warnOnUnregistered: false
+    });
+
+    Table = require('../../lib/bigtable/table.js');
+  });
+
+  after(function() {
+    mockery.deregisterAll();
+    mockery.disable();
+  });
+
+  beforeEach(function() {
+    table = new Table(BIGTABLE, TABLE_ID);
+  });
+
+  afterEach(function() {
+    sinon.restore();
+
+    Object.keys(FakeMutation).forEach(function(spy) {
+      if (FakeMutation[spy].reset) {
+        FakeMutation[spy].reset();
+      }
+    });
+
+    FakeFilter.parse.reset();
+  });
+
+  describe('instantiation', function() {
+    it('should inherit from GrpcServiceObject', function() {
+      var config = table.calledWith_[0];
+
+      assert(table instanceof FakeGrpcServiceObject);
+      assert.strictEqual(config.parent, BIGTABLE);
+      assert.strictEqual(config.id, TABLE_NAME);
+
+      assert.deepEqual(config.methods, {
+        create: true,
+        delete: {
+          protoOpts: {
+            service: 'BigtableTableService',
+            method: 'deleteTable'
+          },
+          reqOpts: {
+            name: TABLE_NAME
+          }
+        },
+        exists: true,
+        get: true,
+        getMetadata: {
+          protoOpts: {
+            service: 'BigtableTableService',
+            method: 'getTable'
+          },
+          reqOpts: {
+            name: TABLE_NAME
+          }
+        }
+      });
+    });
+
+    it('should use Bigtable#createTable to create the table', function(done) {
+      var fakeOptions = {};
+
+      BIGTABLE.createTable = function(name, options, callback) {
+        assert.strictEqual(name, TABLE_ID);
+        assert.strictEqual(options, fakeOptions);
+        callback(null, table); // done
+      };
+
+      table.create(fakeOptions, done);
+    });
+  });
+
+  describe('formatName_', function() {
+    it('should format the table name to include the cluster name', function() {
+      var tableName = Table.formatName_(BIGTABLE.clusterName, TABLE_ID);
+      assert.strictEqual(tableName, TABLE_NAME);
+    });
+
+    it('should not re-format the table name', function() {
+      var tableName = Table.formatName_(BIGTABLE.clusterName, TABLE_NAME);
+      assert.strictEqual(tableName, TABLE_NAME);
+    });
+  });
+
+  describe('formatRowRange_', function() {
+    it('should create a row range object', function() {
+      var fakeRange = {
+        start: 'a',
+        end: 'b'
+      };
+
+      var range = Table.formatRowRange_(fakeRange);
+
+      assert.deepEqual(range, {
+        startKey: 'a',
```

Comment:
This needs to check that the results from calling `Mutation.convertToBytes` is what is used as the start and end keys.